code/game/gamemodes/gang/dominator.dm

```diff
@@ -0,0 +1,225 @@
+/obj/machinery/dominator
+	name = "dominator"
+	desc = "A visibly sinister device. Looks like you can break it if you hit it enough."
+	icon = 'icons/obj/machines/dominator.dmi'
+	icon_state = "dominator"
+	density = 1
+	anchored = 1
+	layer = 3.6
+	luminosity = 2
+	var/maxhealth = 200
+	var/health = 200
+	var/datum/gang/gang
+	var/operating = 0	//-1=broken, 0=standby, 1=takeover
+	var/warned = 0	//if this device has set off the warning at <3 minutes yet
+
+
+/obj/machinery/dominator/examine(mob/user)
+	..()
+	if(operating == -1)
+		user << "<span class='danger'>It looks completely busted.</span>"
+		return
+
+	var/time
+	if(gang && isnum(gang.dom_timer))
+		time = max(gang.dom_timer, 0)
+		if(time > 0)
+			user << "<span class='notice'>Hostile Takeover in progress. Estimated [time] seconds remain.</span>"
+		else
+			user << "<span class='notice'>Hostile Takeover of [station_name()] successful. Have a great day.</span>"
+	else
+		user << "<span class='notice'>System on standby.</span>"
+	user << "<span class='danger'>System Integrity: [round((health/maxhealth)*100,1)]%</span>"
+
+/obj/machinery/dominator/process()
+	..()
+	if(gang && isnum(gang.dom_timer))
+		if(gang.dom_timer > 0)
+			playsound(loc, 'sound/items/timer.ogg', 10, 0)
+			if(!warned && (gang.dom_timer < 180))
+				warned = 1
+				var/area/domloc = get_area(loc)
+				gang.message_gangtools("Less than 3 minutes remain in hostile takeover. Defend your dominator at [initial(domloc.name)]!")
+				for(var/datum/gang/G in ticker.mode.gangs)
+					if(G != gang)
+						G.message_gangtools("WARNING: [gang.name] Gang takeover imminent. Their dominator at [initial(domloc.name)] must be destroyed!",1,1)
+		else
+			processing_objects.Add(src)
+
+/obj/machinery/dominator/proc/healthcheck(damage)
+	var/iconname = "dominator"
+	if(gang)
+		iconname += "-[gang.color]"
+		luminosity = 3
+
+	var/datum/effect/system/spark_spread/sparks = new /datum/effect/system/spark_spread
```

Comment:
You're making a new spark system every time, you could just store it on an object variable...